dataservice/extensions.py

```diff
@@ -2,6 +2,59 @@
 from flask_sqlalchemy import SQLAlchemy, Model
 
 
-db = SQLAlchemy()
+class CRUDMixin(Model):
+    """
+    Mixin that adds convenience methods for CRUD
+    (create, read, update, delete) operations.
+    """
+
+    @classmethod
+    def create(cls, **kwargs):
+        """
+        Create a new record and save it the database.
+        """
+        instance = cls(**kwargs)
+        return instance.save()
+
+    def update(self, commit=True, **kwargs):
+        """
+        Update specific fields of a record.
+        """
+        for attr, value in kwargs.items():
+            setattr(self, attr, value)
+        return commit and self.save() or self
+
+    @classmethod
+    def find_all(cls, **kwargs):
+        """
+        Find all records, filter by kwargs
+        """
+        return [instance for instance in cls.query.filter_by(**kwargs)]
```

Comment:
Can get rid of the comprehension and use `all()`

```cls.query.filter_by(...).all()```

But probably best to replace this with pagination or `yield_per` if we really need to.

Reply:
Yep, agreed.